### DISCUSSION TITLE
Why are Snapshot node "database" and "schema" configs special?

### DISCUSSION THREAD

**Discussion by EliasAthey** [dbt-labs/dbt-core#9774] [Q&A]
Related function in parser code: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/parser/base.py#L286

Why are snapshots handled differently than other node types when parsing the relation name?

Our project has a custom generate_schema_name macro. I'm trying to materialize my snapshots in the same fashion as models (which automatically use our custom macro) so that snapshots are target-dependent, but am not succeeding. I want to know more about the decision to make snapshots "special". Thanks!

**Comment by graciegoheen** [comment]
This is fixed as of `v1.9` -> https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.9#snapshots-improvements

> Standard schema and database configs supported: Snapshots will now be consistent with other dbt resource types. You can specify where environment-aware snapshots should be stored.
